From d45c8bd853de1aa99c816591bd0a0cba147ba7b8 Mon Sep 17 00:00:00 2001
From: "Vishnu Swaroop Kumar Sarma, Duddu" <duddu.swaroop@intel.com>
Date: Fri, 12 Jul 2019 22:29:37 +0530
Subject: [PATCH] UGW_SW-40557 fill subif_data only when dev is NULL

For dsl ATM/NAS interface during regsitration the netdev may come NULL.
Fill subif_data only for this case, else the registration for ptm vlan interface fails
---
 drivers/net/datapath/dpm/datapath_misc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/datapath/dpm/datapath_misc.c b/drivers/net/datapath/dpm/datapath_misc.c
index 01ebfd5d345c..67a7f9957d69 100644
--- a/drivers/net/datapath/dpm/datapath_misc.c
+++ b/drivers/net/datapath/dpm/datapath_misc.c
@@ -1320,7 +1320,7 @@ int32_t dp_sync_subifid(struct net_device *dev, char *subif_name,
 	port = get_dp_port_info(0, subif_id->port_id);
 	if (!port)
 		return DP_FAILURE;
-	if (port->alloc_flags & DP_F_FAST_DSL)
+	if (port->alloc_flags & DP_F_FAST_DSL && dev == NULL)
 		subif_data = (void *)subif_name;
 	/*check flag for register / deregister to update/del */
 	if (flags & DP_F_DEREGISTER) {
@@ -1364,7 +1364,7 @@ int32_t dp_sync_subifid_priv(struct net_device *dev, char *subif_name,
 	port = get_dp_port_info(0, subif_id->port_id);
 	if (!port)
 		return DP_FAILURE;
-	if (port->alloc_flags & DP_F_FAST_DSL)
+	if (port->alloc_flags & DP_F_FAST_DSL && dev == NULL)
 		subif_data = (void *)subif_name;
 	/*check flag for register / deregister to update/del */
 	if (flags & DP_F_DEREGISTER) {
